sql/core/src/test/scala/org/apache/spark/sql/StringLiteralCoalescingSuite.scala

```diff
@@ -0,0 +1,739 @@
+ */
+
+package org.apache.spark.sql
+
+import org.apache.spark.sql.test.SharedSparkSession
+
+/**
+ * Test suite for string literal coalescing across all SQL grammar contexts.
+ *
+ * String literal coalescing allows multiple consecutive string literals to be
+ * automatically concatenated: 'hello' 'world' becomes 'helloworld'.
+ *
+ * This feature works in all contexts where string literals are accepted,
+ * not just in expressions.
+ */
+class StringLiteralCoalescingSuite extends QueryTest with SharedSparkSession {
+
+  // ========================================================================
+  // Basic String Literal Coalescing Tests
+  // ========================================================================
+
+  test("string coalescing in SELECT expressions") {
+    checkAnswer(
+      sql("SELECT 'hello' 'world'"),
+      Row("helloworld")
+    )
+
+    checkAnswer(
+      sql("SELECT 'one' 'two' 'three'"),
+      Row("onetwothree")
+    )
+  }
+
+  test("string coalescing with mixed quote styles") {
+    checkAnswer(
+      sql("SELECT 'hello' \"world\""),
+      Row("helloworld")
+    )
+
+    checkAnswer(
+      sql("SELECT \"hello\" 'world' \"!\""),
+      Row("helloworld!")
+    )
+  }
+
+  // ========================================================================
+  // DDL Context Coalescing Tests
+  // ========================================================================
+
+  test("string coalescing in DDL LOCATION clause") {
+    withTempPath { dir =>
+      withTable("t") {
+        val path1 = dir.getAbsolutePath
+        val path2 = "/test_data"
+        sql(s"CREATE TABLE t (id INT) USING parquet LOCATION '$path1' '$path2'")
+
+        val location = sql("DESCRIBE EXTENDED t")
+          .filter("col_name = 'Location'")
+          .select("data_type")
+          .collect()
+          .head
+          .getString(0)
+
+        assert(location.contains(path1 + path2))
+      }
+    }
+  }
+
+  test("string coalescing in DDL COMMENT clause") {
+    withTable("t") {
+      sql("CREATE TABLE t (id INT) COMMENT 'This is ' 'a multi' 'part comment'")
+
+      val comment = sql("DESCRIBE EXTENDED t")
+        .filter("col_name = 'Comment'")
+        .select("data_type")
+        .collect()
+        .head
+        .getString(0)
+
+      assert(comment == "This is a multipart comment")
+    }
+  }
+
+  test("string coalescing in column COMMENT") {
+    withTable("t") {
+      sql("CREATE TABLE t (id INT COMMENT 'User ' 'ID' ' number')")
+
+      val comment = sql("DESCRIBE t")
+        .filter("col_name = 'id'")
+        .select("comment")
+        .collect()
+        .head
+        .getString(0)
+
+      assert(comment == "User ID number")
+    }
+  }
+
+  // ========================================================================
+  // LIKE and Pattern Matching Tests
+  // ========================================================================
+
+  test("string coalescing in LIKE pattern") {
+    checkAnswer(
```

Comment:
There are many tests of this format, with
```
test("category") {
  checkAnswer(
    sql("some query"),
    Row(true)
  )
}
```

It seems like we could simplify this a bunch by just putting a big `Seq` of strings and then doing a `.foreach` on it and calling `checkAnswer(sql(query), Row(true))` on it. We can still port the existing test case names as comments and put them in comments in the `Seq` instead.